pynamodb/connection/base.py

```diff
@@ -349,124 +349,7 @@ def send_pre_boto_callback(self, operation_name, req_uuid, table_name):
             log.exception("pre_boto callback threw an exception.")
 
     def _make_api_call(self, operation_name: str, operation_kwargs: Dict, settings: OperationSettings = OperationSettings.default) -> Dict:
```

Comment:
are we concerned about `settings: OperationSettings` now not being used? I see it only passed into `self._create_prepared_request`

Reply:
ah thats probably `Reimplement extra_headers (through the before-sign. hook)`

Reply:
The motivation for 'settings' was to allow per-request controls of retries, e.g. you might want to hedged-retry a read but not a write. Honestly didn't end up using this anywhere. With retries delegated to boto, this might not be needed since we'd no longer retry perm-fails indiscriminately.

Reply:
I removed this in https://github.com/pynamodb/PynamoDB/pull/1079/commits/4f28c9f2a43ea57f45b291580a194121b9363110 since I can't find a good way to retain this without taking more deps on botocore internals, and I think it's unlikely to be used much